internal/resolution/variablesources/bundle.go

```diff
@@ -2,62 +2,36 @@ package variablesources
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sort"
 
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
-	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
 	catalogsort "github.com/operator-framework/operator-controller/internal/catalogmetadata/sort"
 	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
 )
 
-var _ input.VariableSource = &BundlesAndDepsVariableSource{}
-
-type BundlesAndDepsVariableSource struct {
-	catalogClient   BundleProvider
-	variableSources []input.VariableSource
-}
-
-func NewBundlesAndDepsVariableSource(catalogClient BundleProvider, inputVariableSources ...input.VariableSource) *BundlesAndDepsVariableSource {
-	return &BundlesAndDepsVariableSource{
-		catalogClient:   catalogClient,
-		variableSources: inputVariableSources,
-	}
-}
-
-func (b *BundlesAndDepsVariableSource) GetVariables(ctx context.Context) ([]deppy.Variable, error) {
-	var variables []deppy.Variable
-
-	// extract required package variables
-	for _, variableSource := range b.variableSources {
-		inputVariables, err := variableSource.GetVariables(ctx)
-		if err != nil {
-			return nil, err
-		}
-		variables = append(variables, inputVariables...)
-	}
-
-	// create bundle queue for dependency resolution
+func MakeBundleVariables(
+	allBundles []*catalogmetadata.Bundle,
+	requiredPackages []*olmvariables.RequiredPackageVariable,
+	installedPackages []*olmvariables.InstalledPackageVariable,
+) ([]*olmvariables.BundleVariable, error) {
 	var bundleQueue []*catalogmetadata.Bundle
-	for _, variable := range variables {
-		switch v := variable.(type) {
-		case *olmvariables.RequiredPackageVariable:
-			bundleQueue = append(bundleQueue, v.Bundles()...)
-		case *olmvariables.InstalledPackageVariable:
-			bundleQueue = append(bundleQueue, v.Bundles()...)
-		}
+	for _, variable := range requiredPackages {
+		bundleQueue = append(bundleQueue, variable.Bundles()...)
 	}
-
-	allBundles, err := b.catalogClient.Bundles(ctx)
-	if err != nil {
-		return nil, err
+	for _, variable := range installedPackages {
+		bundleQueue = append(bundleQueue, variable.Bundles()...)
 	}
 
 	// build bundle and dependency variables
+	var result []*olmvariables.BundleVariable
```

Comment:
We expect result to always need capacity greater than or equal to length of requiredPackages, right? And (for now at least), also installedPackages?
```suggestion
	result := make([]*olmvariables.BundleVariable, 0, len(requiredPackages) + len(installedPackages))
```

Reply:
> We expect result to always need capacity greater than or equal to length of requiredPackages, right? And (for now at least), also installedPackages?

That is correct, but there is no direct relation between size of these two slices and the final slice (we go through dependencies recursively).

I think here it is better to let Go use default algorithm for expanding the slice.

Aslo https://github.com/operator-framework/operator-controller/pull/498#discussion_r1380172620 and the next comment in this thread. I think it is very similar

Reply:
Yep similar question, similar response: https://github.com/operator-framework/operator-controller/pull/498#discussion_r1380445055

Reply:
Updated: https://github.com/operator-framework/operator-controller/pull/498#discussion_r1380489544